### PR TITLE
MM-55006 Validate team access before returning deleted teams

### PR DESCRIPTION
**Pull request by sbishel**
#### Summary
Add missing permissions check to API endpoint

#### Ticket Link

Fixes https://mattermost.atlassian.net/browse/MM-55006

#### Release Note
```release-note
Add permission check to deleted channel endpoint. 
```

### CODE REVIEW

File: server/channels/api4/channel.go [hunk 1]
```diff
@@ -836,6 +836,11 @@ func getDeletedChannelsForTeam(c *Context, w http.ResponseWriter, r *http.Reques
 		return
 	}
 
+	if !c.App.SessionHasPermissionToTeam(*c.AppContext.Session(), c.Params.TeamId, model.PermissionReadPublicChannel) {
```
Comment (agnivade) [inline server/channels/api4/channel.go:839]: Trying to understand why you chose `model.PermissionReadPublicChannel` and not `PermissionListTeamChannels`. The earlier seems to be specific to a particular channel.

Comment (sbishel) [reply server/channels/api4/channel.go]: Not sure, I probably wasn't.  Updated.

**Review by agnivade — APPROVED** [review]

**Review by esarafianou — APPROVED** [review]

**Review by lindy65 — APPROVED** [review]
Thanks @sbishel - looks good to me 👍

**Pull request merged**